src/vl-map-search.js

```diff
@@ -123,50 +125,117 @@ export class VlMapSearch extends vlElement(HTMLElement) {
+    fetch(this.searchUrl + searchValue).then((response) => {
+      return response.json();
+    }).then((data) => {
+      if (data && data.SuggestionResult) {
+        const resultaten = data.SuggestionResult.map((resultaat) => {
+          return {
+            value: resultaat,
+            label: resultaat,
+          };
+        });
+        this._selectElement.choices = resultaten;
+      }
+    });
+  }
+
+  _searchChoicesByLambertCoördinaat(lambertCoordinaat) {
+    fetch(this._findLocationsByLambertCoordinaatUrl(lambertCoordinaat)).then((response) => {
+      return response.json();
+    }).then((data) => {
+      const choices = [{
+        value: lambertCoordinaat,
+        label: 'Lambert-coördinaat: ' + lambertCoordinaat.toString(),
+      }];
+
+      if (data && data.LocationResult) {
+        data.LocationResult.map((locationResult) => {
+          return {
+            value: locationResult,
+            label: locationResult.FormattedAddress,
+          };
+        }).forEach((choice) => {
+          choices.push(choice);
+        });
+      }
+      this._selectElement.choices = choices;
+    });
+  }
+
   _addChoiceEventListener() {
     if (!this.__choiceEventListenerRegistered) {
       this.__choiceEventListenerRegistered = true;
       this.__choiceEventListener = this._selectElement.addEventListener('choice', (event) => {
         if (event && event.detail && event.detail.choice) {
-          fetch(this.locationUrl + event.detail.choice.value).then((response) => {
-            return response.json();
-          }).then((data) => {
-            if (data && data.LocationResult) {
-              if (this._onSelect) {
-                this._onSelect(data);
-              } else if (this._map) {
-                this._map.zoomTo([
-                  data.LocationResult[0].BoundingBox.LowerLeft.X_Lambert72,
-                  data.LocationResult[0].BoundingBox.LowerLeft.Y_Lambert72,
-                  data.LocationResult[0].BoundingBox.UpperRight.X_Lambert72,
-                  data.LocationResult[0].BoundingBox.UpperRight.Y_Lambert72,
-                ]);
-              }
-            }
-          });
+          const value = event.detail.choice.value;
+
+          if (this._isLambertCoördinaat(value)) {
+            this._zoomToLambertCoördinaat(value);
+          } else if (value instanceof Object) {
+            this._zoomToLocation(value);
+          } else {
+            this._searchAndZoomToLocation(value);
+          }
         }
       });
     }
   }
 
+  _isLambertCoördinaat(value) {
+    // TODO stefanborghys: 28/01/21 prefer to use instanceof but could not import LambertCoordinaat in the test and replaced it for now
+    // instanceof LambertCoordinaat
+    return value.x !== undefined && value.y !== undefined;
+  }
+
+  _zoomToLambertCoördinaat(lambertCoordinaat) {
+    this._map.zoomToGeometry({
+      type: 'Point',
+      coordinates: [lambertCoordinaat.x, lambertCoordinaat.y],
+    }, 10);
```

Comment:
Lijkt me artificieel om een geometry aan te maken, zou dan eerder een bounding box maken op basis van het Lambert coördinaat.

Reply:
Ik heb dit even als unresolved gemarkeerd om te informeren. De code lijkt niet gewijzigd waardoor de feedback nog van toepassing is. @stefanborghys kan jij een kleine update geven?

Reply:
Momenteel is dit dus geworden en is de ontdubbeling van de zoom methode ongedaan gemaakt.
<img width="560" alt="afbeelding" src="https://user-images.githubusercontent.com/1375279/107219688-b777c480-6a11-11eb-8f0a-aa9038a6f2aa.png">
Los ik daarmee dit issue op of moet er nog meer gebeuren.
Had deze dan weer geresolved omdat er was overeengekomen om de methode hernoemen wat bij deze is teruggedraaid.

Reply:
Ik denk dat de functie implementatie nu als volgt geschreven kan worden:

``` JS
this._map.zoomTo([lambertCoordinaat.x, lambertCoordinaat.y];
```

Reply:
Dat lijkt niet te werken.
Als ik hem toepas en probeer in de demo krijg ik dit:
<img width="765" alt="afbeelding" src="https://user-images.githubusercontent.com/1375279/107236235-ca948f80-6a25-11eb-8022-ee9077968f59.png">

Reply:
volgens mij is dit dan ook in orde?

Reply:
Inderdaad in orde as is.